src/libutil/unix/environment-variables.cc

```diff
@@ -0,0 +1,21 @@
+#include "util.hh"
+#include "environment-variables.hh"
+
+extern char * * environ __attribute__((weak));
+
+namespace nix {
+
+void clearEnv()
+{
+    for (auto & name : getEnv())
+        unsetenv(name.first.c_str());
+}
+
+void replaceEnv(const std::map<std::string, std::string> & newEnv)
```

Comment:
I think this function is only used in `processes.cc` so I think it might even make more sense to just put it in that file.

Reply:
I think I want to keep it here because of where it is declared, for now, but yes it does seem that all the env var modification stuff (`clearEnv` too) is just used for process spawning. Once we get that working on Windows it would be good time to revisit this :).